pydra/engine/specs.py

```diff
@@ -445,47 +435,36 @@ class ShellOutSpec:
     def collect_additional_outputs(self, inputs, output_dir, outputs):
         """Collect additional outputs from shelltask output_spec."""
         additional_out = {}
-        for fld in attr_fields(self):
-            if fld.name not in ["return_code", "stdout", "stderr"]:
-                if fld.type in [
-                    File,
-                    MultiOutputFile,
-                    Directory,
-                    int,
-                    float,
-                    bool,
-                    str,
-                    list,
-                ]:
-                    # assuming that field should have either default or metadata, but not both
-                    if (
-                        fld.default is None or fld.default == attr.NOTHING
-                    ) and not fld.metadata:  # TODO: is it right?
-                        raise AttributeError(
-                            "File has to have default value or metadata"
-                        )
-                    elif fld.default != attr.NOTHING:
-                        additional_out[fld.name] = self._field_defaultvalue(
-                            fld, output_dir
-                        )
-                    elif fld.metadata:
-                        if (
-                            fld.type in [int, float, bool, str, list]
-                            and "callable" not in fld.metadata
-                        ):
-                            raise AttributeError(
-                                f"{fld.type} has to have a callable in metadata"
-                            )
-                        else:
-                            additional_out[fld.name] = self._field_metadata(
-                                fld, inputs, output_dir, outputs
-                            )
-                #                        else:
-                #                            additional_out[fld.name] = self._field_metadata(
-                #                                fld, inputs, output_dir, outputs
-                #                            )
-                else:
-                    raise Exception("not implemented (collect_additional_output)")
+        for fld in attr_fields(self, exclude_names=("return_code", "stdout", "stderr")):
+            if fld.type not in [
```

Comment:
i'm actually not sure if we should be checking this here.. but can think about this in a different pr